src/components/GolfDetailsCard/GolfDetailsCard.tsx

```diff
@@ -0,0 +1,47 @@
+import {GolfProps} from "../../services/types.ts";
+
+import "./GolfDetailsCard.css";
+import {GolfDetailsField} from "../GolfDetailsField/GolfDetailsField.tsx";
+import SecondaryButton from "../Buttons/SecondaryButton/SecondaryButton.tsx";
+
+import EditIcon from "../../assets/svg/icons/edit.svg";
+import {Modal} from "../Modal/Modal.tsx";
+import {EditGolfForm} from "../EditGolfForm/EditGolfForm.tsx";
+import {useState} from "react";
+import {updateGolf} from "../../services/courseService.ts";
+
+interface GolfDetailsCardProps {
+    courseDetails: GolfProps;
+}
+
+export function GolfDetailsCard({courseDetails}: GolfDetailsCardProps) {
+    const [isModalVisible, setModalVisible] = useState(false);
+    const [golfDetails, setGolfDetails] = useState(courseDetails);
+
+    const handleSave = async (updatedDetails: GolfProps) => {
+        setModalVisible(false);
+        await updateGolf(updatedDetails.id, updatedDetails.name, updatedDetails.city, updatedDetails.postalCode, updatedDetails.googleMapLinks);
+        setGolfDetails(updatedDetails);
+    };
+
+    return (
+        <div className="golf-details-cards-container">
```

Comment:
consider implementing a progress indicator across your application, for when your waiting for your API responses to come back.